tests/tools.py

```diff
@@ -333,6 +334,23 @@ def emulate_raises_regex(self, exception_type, regex, function, *args, **kwargs)
                     exception_type, regex, str(e)))
         self.fail("No exception was thrown.  Expected exception {0} matching {1}".format(exception_type, regex))
 
+    def emulate_assertDictEqual(self, first, second, msg=None):
```

Comment:
A bare bones implementation not even close to the assert in >= 2.7 (in terms of debugging test failures). We can improve it as needed.

Reply:
Thanks for adding this!